### PR TITLE
Update travis config

### PR DESCRIPTION
**Pull request by davidanthoff**
I think I got it right, but not sure :) Lets see whether tests pass. If this works, we could switch off the appveyor stuff.

### CODE REVIEW

**Comment by fredrikekre** [comment]
You are not allowed to use `matrix` and `jobs` anymore, need to combine them.

**Comment by davidanthoff** [comment]
Hm, I don't understand why this doesn't create 32 bit Windows builds? And at least in the UI right now it also doesn't seem to show allowed failure rows anymore? Weird...

File: .travis.yml [hunk 1]
```diff
@@ -3,14 +3,20 @@ language: julia
 os:
   - linux
   - osx
+  - windows
 
 julia:
   - 1.0
   - 1.1
```
Comment (KristofferC) [inline .travis.yml]: While changing this, we might as well remove 1.1. and 1.2.

File: .travis.yml [hunk 2]
```diff
@@ -19,16 +25,20 @@ branches:
...
         - julia --project=docs/ docs/make.jl
       name: "HTML"
       after_success: skip
-
-after_success:
-  - julia -e 'import Pkg; Pkg.add("Coverage"); using Coverage; Codecov.submit(Codecov.process_folder())'
+  exclude:
+  - os: osx
+    arch: x86
+  allow_failures:
```
Comment (KristofferC) [inline .travis.yml]: This should be removed imo. We want this package to work on nightly and some red crosses might make someone annoyed enough to look into it.

**Comment by timholy** [comment]
Seems fine to me, but I won't merge since I'm unsure whether you're still working on the 32-bit Windows issue.

**Comment by davidanthoff** [comment]
I don't really understand why the 32 bit stuff doesn't work and don't have any more ideas. I asked on Slack, but I guess we could merge this in the meantime and always do another PR if someone figures this out.

**Comment by timholy** [comment]
I don't seem to have access to the AppVeyor settings. @KristofferC, can you shut it off?

**Comment by davidanthoff** [comment]
Maybe we wait with shutting appveyor off until we have 32 bit windows build working on travis?

**Comment by KristofferC** [comment]
Yeah, I think this https://github.com/JuliaDebug/JuliaInterpreter.jl/blob/a0e209e046c8ad650525714c2117f26024a8ec44/src/optimize.jl#L333-L337 code path is only hit on 32 bit Windows so probably want to still test that.

**Pull request merged**